### DISCUSSION TITLE
Custom name of primary key (add primary props for field)

### DISCUSSION THREAD

**Discussion by alexdeia** [payloadcms/payload#5683] [Q&A]
Hello.
I read the docs https://payloadcms.com/docs/fields/overview#customizable-id about it, but it only describes changing the type, not the name. How can I do it?

**Expectation:**
Possible to rename `id` field to, for example, `identity` and this field will become the new primary key.

```typescript
...
  fields: [
    {
      name: 'identity',
      type: 'text',
      label: 'Identity',
      unique: true,
      required: true,
      primary: true, // -> THIS for example
    },
    {
      name: 'description',
      type: 'textarea',
      label: 'Description',
    },
  ],
...
```

**Comment by arvenom** [comment]
got the same issue! Help would be nice here.

**Comment by takayumi** [comment]
I can't seem to change the custom ID for MongoDB. Have you found a workaround @alexdeia?

**Reply by alexdeia** [reply]
I don't use MongoDB, PostgreSQL only.

Unfortunately, I've resigned myself to working with what I've got.
